manifests/init.pp

```diff
@@ -74,7 +77,18 @@
   $yaml_fact_path_real = pick($yaml_fact_path, "${confdir}/facts.yaml")
   $server_config_file_real = pick($server_config_file, "${confdir}/server.cfg")
   $client_config_file_real = pick($client_config_file, "${confdir}/client.cfg")
-  $ssl_client_certs_dir_real = pick($ssl_client_certs_dir, "${confdir}/clients")
+
+  $ssl_client_certs_dir_real = pick($ssl_client_certs_dir, "${confdir}/ssl/clients")
+  $ssl_server_public_path    = "${confdir}/ssl/server_public.pem"
+  $ssl_server_private_path   = "${confdir}/ssl/server_private.pem"
+
+  $middleware_ssl_ca_real   = pick_default($middleware_ssl_ca, $ssl_ca_cert)
+  $middleware_ssl_cert_real = pick_default($middleware_ssl_cert, $ssl_server_public)
+  $middleware_ssl_key_real  = pick_default($middleware_ssl_key, $ssl_server_private)
```

Comment:
wh00t! i was about to comment here that we should use this instead of the horrendous if/else!
thank you so much :D